lib/features/email/presentation/controller/single_email_controller.dart

```diff
@@ -1772,4 +1707,56 @@ class SingleEmailController extends BaseController with AppLoaderMixin {
         AppLocalizations.of(currentContext!).eventReplyWasSentUnsuccessfully);
     }
   }
+
+  void handleDownloadAttachmentAction(BuildContext context, Attachment attachment) {
+    if (PlatformInfo.isWeb) {
+      downloadAttachmentForWeb(attachment);
+    } else if (PlatformInfo.isMobile) {
+      exportAttachment(context, attachment);
+    } else {
+      log('EmailView::handleDownloadAttachmentAction: THE PLATFORM IS SUPPORTED');
+    }
+  }
+
+  void handleViewAttachmentAction(BuildContext context, Attachment attachment) {
+    if (PlatformInfo.isWeb) {
+      if (PlatformInfo.isCanvasKit && attachment.isDisplayedPDFIcon) {
+        previewPDFFileAction(context, attachment);
+      } else {
+        downloadAttachmentForWeb(attachment);
+      }
+    } else if (PlatformInfo.isMobile) {
+      exportAttachment(context, attachment);
+    } else {
+      log('EmailView::_handleViewAttachmentAction: THE PLATFORM IS SUPPORTED');
+    }
+  }
+
+  Future<void> previewPDFFileAction(BuildContext context, Attachment attachment) async {
+    final accountId = mailboxDashBoardController.accountId.value;
+    final downloadUrl = mailboxDashBoardController.sessionCurrent
+      ?.getDownloadUrl(jmapUrl: dynamicUrlInterceptors.jmapUrl);
+
+    if (accountId == null || downloadUrl == null) {
+      appToast.showToastErrorMessage(
+        context,
+        AppLocalizations.of(context).noPreviewAvailable);
+      return;
+    }
+
+    await Get.generalDialog(
+      barrierColor: Colors.black.withOpacity(0.8),
+      pageBuilder: (_, __, ___) {
+        return PointerInterceptor(
+          child: PDFViewer(
```

Comment:
How to set up this viewer more details in language? Because in error case, this viewer show error in weird language

Reply:
We have support for many languages.